src/eval_llvm.cpp

```diff
@@ -1,41 +1,103 @@
+ *  $X      uint64_t      `4d2`              Hexadecimal number (64 bit)
+ * --------------------------------------------------------------------------
+ *  $s      const char *  `foo`              Zero-terminated string
+ * --------------------------------------------------------------------------
+ *  $t      Variable      `i1`               Scalar variable type
+ *  $T      Variable      `<8 x i1>`         Vector variable type
+ *  $h      Variable      `f32`              Short type abbreviation
+ * --------------------------------------------------------------------------
+ *  $m      Variable      `i8`               Scalar variable type
+ *                                           (masks promoted to 8 bits)
+ *  $M      Variable      `<8 x i8>`         Vector variable type
+ *                                           (masks promoted to 8 bits)
+ * --------------------------------------------------------------------------
+ *  $v      Variable      `%p1234`           Variable name
+ * --------------------------------------------------------------------------
+ *  $V      Variable      `<8 x i1> %p1234`  Type-qualified vector var. name
+ * --------------------------------------------------------------------------
+ *  $a      Variable      `1`                Scalar variable alignment
+ *  $A      Variable      `16`               Vector variable alignment
+ * --------------------------------------------------------------------------
+ *  $o      Variable      `5`                Variable offset in param. array
+ * --------------------------------------------------------------------------
+ *  $l      Variable      `1`                Literal value of variable
+ * --------------------------------------------------------------------------
+ *  $w      (none)        `16`               Vector width of LLVM backend
+ *  $z      (none)        `zeroinitializer`  Zero initializer string
+ * --------------------------------------------------------------------------
+ *
+ * Pointers should be wrapped in braces, as in `{i8*}` or `{$t*}`. This will
+ * allow them to be replaced by the opaque `ptr` type on newer versions of LLVM
+ * that use this convention. An extended form of this syntax `{a|b}` causes `a`
+ * and `b` to be generated for LLVM with non-opaque and opaque pointers,
+ * respectively.
+ *
+ * Another syntax pattern used in a few places is `$<foo$>`. It expands to
+ * `foo` at the top level and `<16 x foo>` when the generated code is part
+ * of a subroutine (where 16 is the vector width in this example).
+ */
+
 #include "eval.h"
 #include "internal.h"
 #include "log.h"
 #include "var.h"
 #include "vcall.h"
 #include "op.h"
 
+#define put(...)                                                               \
+    buffer.put(__VA_ARGS__)
+
+#define fmt(fmt, ...)                                                          \
+    buffer.fmt_llvm(count_args(__VA_ARGS__), fmt, ##__VA_ARGS__)
+
```

Comment:
Write those macros on a single line?

Reply:
One seems too long- this is what clang-format generated, and i broke the short one into two lines for consistency. Or did I misunderstand?